sympy/assumptions/assume.py

```diff
@@ -86,80 +131,217 @@ def arg(self):
         x + 1
 
         """
-        return self._args[1]
+        args = self.args
+        if len(args) == 1:
+            return args[0]
+        return args
 
     @property
     def args(self):
-        return self._args[1:]
+        return self._args[1]
 
     @property
     def func(self):
         return self._args[0]
 
     @cacheit
     def sort_key(self, order=None):
-        return (self.class_key(), (2, (self.func.name, self.arg.sort_key())),
+        return (self.class_key(),
+                (2, (self.func.sort_key(), self.arg.sort_key())),
                 S.One.sort_key(), S.One)
 
-    def __eq__(self, other):
-        if type(other) is AppliedPredicate:
-            return self._args == other._args
-        return False
-
-    def __hash__(self):
-        return super().__hash__()
-
     def _eval_ask(self, assumptions):
-        return self.func.eval(self.arg, assumptions)
+        return self.func.eval(self.args, assumptions)
 
     @property
     def binary_symbols(self):
         from sympy.core.relational import Eq, Ne
-        if self.func.name in ['is_true', 'is_false']:
+        if self.func.name.name in ['is_true', 'is_false']:
```

Comment:
Should this be something like
```
if self.func in [Q.is_true, Q.is_false]:
```
?

Reply:
Turned out that `Q.is_false` does not exist at all. I will fix this...